metricflow/api/simple_api.py

```diff
@@ -0,0 +1,259 @@
+from typing import Dict, List, Optional
+
+from metricflow.configuration.config_handler import ConfigHandler
+from metricflow.configuration.constants import CONFIG_DWH_SCHEMA
+from metricflow.dataflow.sql_table import SqlTable
+from metricflow.engine.metricflow_engine import (
+    MetricFlowEngine,
+    MetricFlowExplainResult,
+    MetricFlowQueryRequest,
+    MetricFlowQueryResult,
+)
+from metricflow.engine.models import Dimension, Materialization, Metric
+from metricflow.engine.utils import build_user_configured_model_from_config, convert_to_datetime
+from metricflow.model.model_validator import ModelValidator
+from metricflow.model.objects.user_configured_model import UserConfiguredModel
+from metricflow.model.semantic_model import SemanticModel
+from metricflow.model.validations.validator_helpers import ValidationIssueType
+from metricflow.protocols.sql_client import SqlClient
+from metricflow.sql.optimizer.optimization_levels import SqlQueryOptimizationLevel
+from metricflow.sql_clients.sql_utils import make_sql_client_from_config
+
+
+class MetricFlowSimpleAPI:
```

Comment:
Let's get rid of the `SimpleAPI` appellation here. Simple because it leaves me wondering if there's a Complicated one someplace, and API because that's more rightly a reference to all of the classes of this sort.
 
I think something like `MetricFlowClient` might be the right name for this class. It's the in-code entrypoint to MetricFlow as a whole. All you need is a defined model and a warehouse and then you can issue requests via the methods defined below.

Another option is to define a Protocol called `MetricFlowClient` that has the base set of public method signatures specified, and then to name this something like `ConfigInitializedMetricFlowClient` and remove the optional parameters from the initializer. This streamlines the initializer here, makes it obvious  and allows other developers the option of implementing their own custom MetricFlowClient classes.

Reply:
changed to MetricFlowClient